home-manager.nix

```diff
@@ -332,16 +404,31 @@ in
                     fi
                 done
             fi
+
+            # If it is a directory and it's empty
+            if [ -d ${mountPoint} ] && [ -z "$(ls -A ${mountPoint})" ]; then
+              echo "Removing empty directory ${mountPoint}"
+              rm -d ${mountPoint}
+            fi
           '';
 
-        mkUnmountsForPath = persistentStoragePath:
+        mkLinkCleanupForPath = persistentStoragePath:
           concatMapStrings
-            (mkUnmount persistentStoragePath)
-            cfg.${persistentStoragePath}.directories;
+            (mkLinkCleanup persistentStoragePath)
+            (map getPath (filter isSymlink cfg.${persistentStoragePath}.directories));
+
```

Comment:
Go home, git. You're drunk.

For the record, I didn't actually remove `mkUnmountsForPath` (it's on line 375 now), git decided to diff it that way.